### ISSUE TITLE
Log training stats

### ISSUE THREAD

**Issue by austinagii** [austinagii/Lumiere#5]
Context:

The current training script logs the following metrics in a progress bar to the command line to track model performance during training:
- Training & validation loss (cross entropy)
- Training & Validation perplexity 
- Gradient norm
- Learning rate 
- Total training time 

Problem: 

With the current ephemeral logging it is difficult to see how training metrics improve (if at all) over time and to spot potential issues with the models configuration. Also, the information being logged is insufficient to form a comprehensive picture of training problems that need attention.

Solution:

Expand the range of metrics being tracked and log those metrics so that they can be visualized. In addition to the above metrics currently being tracked the following should be tracked:
- Gradient norms per layer
- Weight norm per layer 
- Total weight norm

These should be logged per batch and displayed via w&b or tensor board.
